cycler/__init__.py

```diff
@@ -180,21 +201,27 @@ def change_key(self, old, new):
             return
         if new in self._keys:
             raise ValueError(
-                "Can't replace {old} with {new}, {new} is already a key"
-                .format(old=old, new=new)
+                f"Can't replace {old} with {new}, {new} is already a key"
             )
         if old not in self._keys:
-            raise KeyError("Can't replace {old} with {new}, {old} is not a key"
-                           .format(old=old, new=new))
+            raise KeyError(
+                f"Can't replace {old} with {new}, {old} is not a key"
+            )
 
         self._keys.remove(old)
         self._keys.add(new)
 
-        if self._right is not None and old in self._right.keys:
+        if (
+            self._right is not None
+            and isinstance(self._right, Cycler)
+            and old in self._right.keys
+        ):
             self._right.change_key(old, new)
 
         # self._left should always be non-None
         # if self._keys is non-empty.
+        elif self._left is None:
+            pass
```

Comment:
This explicit check is to type narrow for mypy, it should not ever evaluate to `True`, could raise a more informative error message instead of just passing, though I suspect even in cases where it _would_ evaluate to true it will blow up earlier.

Could adjust the comment to reflect that the check has been added